common/compat.c

```diff
@@ -93,8 +93,12 @@
 #include <unistd.h>
 
 #if defined (HAVE_PROGRAM_INVOCATION_SHORT_NAME) && !HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME
+#ifdef __UCLIBC__
+extern const char *program_invocation_short_name;
```

Comment:
The fact that there is a conflicting definition, means that HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME doesn't work well. configure.ac seems to be checking for it in argp.h, while the manpage in fedora documents it in `errno.h`

Reply:
I noticed that while I was trying to fix this. Both musl and uClibc-ng specify program_invocation_short_name in errno.h  . Changing it there did not help.

Reply:
Wouldn't it need defining _GNU_SOURCE prior to including the file?

Reply:
Looks like yes.

Reply:
Would it help if you change the configure check to:
```sh
# Check if these are declared and/or available to link against
AC_CHECK_DECLS([program_invocation_short_name]), [], [], [#include <error.h>])
```

Reply:
I assume you meant errno.h . Added.

Reply:
No, actually.  The condition `#if defined (HAVE_PROGRAM_INVOCATION_SHORT_NAME) && !HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME` means that the symbol is found in the library but the declaration in a header file is missing. If uClibc has a declaration, p11-kit should use it rather than redefining it.

The `AC_LINK_IFELSE` check is for library symbol, so it is not relevant here. Look for `AC_CHECK_DECLS` a few lines above.

Reply:
I think this is what you meant.

Reply:
Thanks for the update, but do we still need the changes in common/compat.c?

Reply:
Looks like no. Will update.